pkg/cli/sanitize/hide_secrets.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanitize
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"helm.sh/helm/v3/pkg/release"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	hiddenSecretValue = "[HIDDEN]"
+)
+
+// HideManifestSecrets sanitizes release manifest and replaces it.
+// Manifest cannot be applied after this operation.
+// Indentation and extra spaces in Secret's `data` and `stringData` sections can be impacted.
+func HideManifestSecrets(r *release.Release) error {
+	if r == nil {
+		return nil
+	}
+	manifest, err := hideSecrets(r.Manifest)
+	if err != nil {
+		return err
+	}
+
+	r.Manifest = manifest
+
+	return nil
+}
+
+// hideSecrets replaces values in Secrets in the chart manifest with
+// `[HIDDEN]` value.
+func hideSecrets(manifest string) (string, error) {
+	resources := strings.Split(manifest, "\n---")
```

Comment:
How would this handle changes in the output string? How would this handle alternative output formats? For example `helm install --output json`? What if the output of `helm show` changed in the future?

I think these assumptions either need to be documented as expected input, or the function needs to be smarter about how it treats incoming string data. e.g. using some kind of string tokenizer like `text/scanner`.

https://pkg.go.dev/text/scanner@go1.16.6